### ISSUE TITLE
Suggestion: store settings in an external config file.

### ISSUE THREAD

**Issue by FiXato** [excid3/logbot#16]
Currently you have to edit the logbot.py file to change settings. As a result, you can't easily run multiple instances of the same script.
It would be nice if the script only used the current constants as defaults, and allowed for a config file to be passed as a commandline argument that would override the defaults.

I wouldn't mind implementing this myself, but as I'm new to Python and don't really know what the config file standard is (YAML, JSON, ini, etc?), maybe we could discuss it in this ticket. :)

**Comment by excid3** [comment]
Since I moved to Rails, I've really preferred using YAML. Python doesn't have a builtin Yaml parser, but it does have an INI parser. Check out this: http://docs.python.org/library/configparser.html

**Comment by FiXato** [comment]
Yeah, YAML is a lovely format that's also one of my favourite formats. JSON would come close though.
ConfigParser looks nice, but it only supports strings and ints by the looks of it. Then again, using spaces as separators would work well enough for keeping the list of channels.

**Comment by excid3** [comment]
I found this for reading/writing arrays in INIs http://stackoverflow.com/questions/4097139/reading-array-from-config-file-in-python

**Comment by FiXato** [comment]
Still not as clean as how they would be shown in YAML ;-)
I'll do some prototyping on a separate branch.

**Comment by FiXato** [comment]
Looks like the project used to have config file support at some point? https://github.com/excid3/logbot/commit/aaa7200fa2c17f03ad8cd7a023ad92159a58d17c

Any specific reason you didn't include that in your (second?) rewrite?

**Comment by excid3** [comment]
Just convenience actually. Was trying to focus on the bot itself rather than this. It felt unnecessary at the time so I didn't readd it. 

Glad you found that. I was looking for it earlier but couldn't remember which project that was in. :)
